src/storage/countCache/InMemory.js

```diff
@@ -2,26 +2,18 @@ import { CONFIGURATIONS, EVENTS,EVENTS_COUNT } from '../../utils/store/constants
 import EventProcessor from "../../events/eventProcessor.js";
 import AsyncLock from 'async-lock';
 
+// TODO: Add synchronization
 const lock = new AsyncLock();
 
-
 class CountCache {
 
     constructor(store) {
         this.store = store;
         const settings = store.get(CONFIGURATIONS);
         const eventsCount = store.get(EVENTS_COUNT);
-
-        this.onFullQueue = false;
         this.maxQueue = settings.size.metricsQueueSize;
-        if (eventsCount && eventsCount.length > 0) {
```

Comment:
This should not be removed because I am checking if there are events in store from client created from previous sdk key and if there are any flush those events..

Reply:
I think you are right. Thanks Mahrukh, I was thinking CountCache is created per sdk.